### DISCUSSION TITLE
Command exit status 127;  /var/spool/slurmd/job16281783/slurm_script: line 1155: touch: command not found

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#3347] [Q&A]
Hi, 
I have a test nextflow script with multiple processes. One of the process is giving me the attached issue.
The same process ran smoothly without issue and then I added an if statement. I think the if statement should not cause the issue.
Any help? Attached is the log and the script.
[mostest_with_nextflow.nf.txt](https://github.com/nextflow-io/nextflow/files/9924616/mostest_with_nextflow.nf.txt)
[touch_command_not_found_nextflow_log.txt](https://github.com/nextflow-io/nextflow/files/9924617/touch_command_not_found_nextflow_log.txt)

**Comment by mribeirodantas** [comment]
The first thing that I spotted is the setting of process directives with an equal symbol. Even though you have to use
```
...
executor='slurm'
...
```
in `nextflow.config`, when it's within a Nextflow process you should just use space. For example:
```
...
executor 'slurm'
...
```

The second thing is that, if possible, I strongly suggest you to port your pipeline code to DSL2, which has several improvements, which includes better readability and debugging.

Can you confirm that `touch` is installed in the environment it's being called?

**Comment by zillurbmb51** [comment]
Hi @mribeirodantas 
Thank you.
The same configuration works perfectly for all other processes (`executor='slurm'`) and even for this process it worked previously.

I am using a system that can not be connected with the internet and I am not sure how to port the pipeline to DSL2.

Yes, `touch` command works. I have used `touch` to create files after the issue without any problem.

**Comment by zillurbmb51** [comment]
I have changed the excecutor from `slurm` to `local` and now I am getting a new error `null command not found`.
Any help?

[null_command_not_found_nextflow.log.txt](https://github.com/nextflow-io/nextflow/files/9927146/null_command_not_found_nextflow.log.txt)

**Reply by mribeirodantas** [reply]
Are you expecting a command called null to be run? Check the `.command.sh` of this task (`/cluster/p/p33/cluster/users/mohammadzr/metabolomics/scripts/work/e2/45278c385db307f2ed78ebd821f9d9`) to see what's being attempted to run.

Are you trying to run something from within `/cluster/p/p33/cluster/users/mohammadzr/metabolomics/scripts/bin`?

**Reply by zillurbmb51** [reply]
I'm not sure why null is attempting to run.

No, I'm not running anything from scrpts/bin.

**Reply by zillurbmb51** [reply]
If I use `executor "slurm"` then I get `error: touch command not found` and if I use `executor local`,  I get error `null command not found` for the same code. Not sure what is wrong here.

**Reply by zillurbmb51** [reply]
Hi again, I tried to cope up with the `dsl2` syntax. I tried with the attached script. Now, I am getting a new error: `Access to unrelate_1.out is undefined since the process does not declare any output`

But I have declared output with path and emit. This same process gives output in dsl1 without any error.

Any help?
[mostest_with_next_dsl2.nf.txt](https://github.com/nextflow-io/nextflow/files/9938061/mostest_with_next_dsl2.nf.txt)

**Reply by mribeirodantas** [reply]
"from" and "into" in channels lines within processes are DSL1. You also need a `workflow` code block in DSL2, which is where you will put the logic (what process will be called, when, how, with what input channels and so on).

You need to check the `.command.sh` file within your task folder (in `work/`). From there, you will understand what exactly is being called. Null can be an issue in your local installation of the `touch` binary, for example.
